chrome/content/zotero/components/tagSelector/tagSelectorList.jsx

```diff
@@ -214,6 +218,84 @@ class TagList extends React.PureComponent {
 			</div>
 		);
 	};
+
+	// If windowing kicks in and the focused tag's node has been removed,
+	// scroll in its direction until it is re-created and focused.
+	async scrollToFocusedTag() {
+		if (this.focusedTagIndex === null) return;
+		if (this.renderedTagRange.includes(this.focusedTagIndex)) return;
+		// Determine the direction of the scrolling
+		let scrollUp = true;
+		if (this.focusedTagIndex > this.renderedTagRange[0]) {
+			scrollUp = false;
+		}
+		let tagsList = document.querySelector('.tag-selector-list');
+		// Sanity check to make sure we're not stuck in infinite loop
+		let maxCounter = 0;
+		// Keep scrolling until the desired tag is rendered
+		while (!this.renderedTagRange.includes(this.focusedTagIndex) && maxCounter < 10) {
+			if (scrollUp) {
+				tagsList.scrollTop -= tagsList.clientHeight;
+			}
+			else {
+				tagsList.scrollTop += tagsList.clientHeight;
+			}
+			maxCounter += 1;
+			// Wait a moment to let this.handleSectionRendered update this.renderedTagRange
+			await Zotero.Promise.delay(10);
+		}
+	}
```

Comment:
This is wildly inefficient. See [Collection.md](https://github.com/bvaughn/react-virtualized/blob/master/docs/Collection.md). We should use the prop `scrollToCell` without the need to track rendered cells, etc. If that is infeasible, we should use `this.positions[this.focusedTagIndex].y` to calculate the scroll position.

Reply:
Oh yes, thank you for pointing me in the direction of `scrollToCell`! This is way better.

Now, when we handle keyboard navigation and the desired tag is not focused, I update `scrollToCell` state to tell `<Collection>`  where to scroll and finish handling the event in `handleSectionRendered` after the desired tag receives focus. What do you think?